payments/guest-wallet/src/main/java/cm/aptoide/pt/guest_wallet/repository/wallet/WalletRepository.kt

```diff
@@ -0,0 +1,36 @@
+package cm.aptoide.pt.guest_wallet.repository.wallet
+
+import cm.aptoide.pt.guest_wallet.repository.wallet.exception.InvalidWalletException
+import cm.aptoide.pt.guest_wallet.repository.wallet.model.WalletResponse
+import cm.aptoide.pt.payment_manager.wallet.domain.WalletData
+import retrofit2.http.GET
+import retrofit2.http.Query
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class WalletRepositoryImpl @Inject constructor(
+  private val walletApi: WalletApi,
+) : WalletRepository {
+
+  @Throws(InvalidWalletException::class)
+  override suspend fun getWallet(id: String): WalletData {
+    val response = walletApi.getWallet(id)
+    return WalletData(
+      address = response.address ?: throw InvalidWalletException("Invalid wallet address"),
```

Comment:
If you will make the `response` fields non-null you will not need to have these guards.
Because trying to access non-nullable field that holds null will throw anyway.
Just to reduce the number of manual checks :-)